tests/conftest.py

```diff
@@ -541,3 +544,114 @@ def _factory(process: Process, state: plumpy.ProcessState = plumpy.ProcessState.
+        *,
+        name: str | None = None,
+        group: str | None = None
+    ) -> None:
+        """Add an entry point.
+
+        :param value: The class or function to register as entry point. The resource needs to be importable, so it can't
+            be inlined. Alternatively, the fully qualified name can be passed as a string.
+        :param entry_point_string: Fully qualified entry point string.
+        :param name: Entry point name.
+        :param group: Entry point group.
+        :returns: The entry point group and name.
+        :raises TypeError: If `entry_point_string`, `group` or `name` are not a string, when defined.
+        :raises ValueError: If `entry_point_string` is not defined, nor a `group` and `name`.
+        :raises ValueError: If `entry_point_string` is not a complete entry point string with group and name.
+        """
+        if not isinstance(value, str):
+            value = f'{value.__module__}:{value.__name__}'
+
+        group, name = self._validate_entry_point(entry_point_string, group, name)
+        entry_point = module_entry_point.EntryPoint(name, value, group)
+        self.eps()[group].append(entry_point)
+
+    @suppress_deprecations
+    def remove(
+        self, entry_point_string: str | None = None, *, name: str | None = None, group: str | None = None
+    ) -> None:
+        """Remove an entry point.
+
+        :param value: Entry point value, fully qualified import path name.
+        :param entry_point_string: Fully qualified entry point string.
+        :param name: Entry point name.
+        :param group: Entry point group.
+        :returns: The entry point group and name.
+        :raises TypeError: If `entry_point_string`, `group` or `name` are not a string, when defined.
+        :raises ValueError: If `entry_point_string` is not defined, nor a `group` and `name`.
+        :raises ValueError: If `entry_point_string` is not a complete entry point string with group and name.
+        """
+        group, name = self._validate_entry_point(entry_point_string, group, name)
+
+        for entry_point in self.eps()[group]:
+            if entry_point.name == name:
+                self.eps()[group].remove(entry_point)
+                break
+        else:
+            raise KeyError(f'entry point `{name}` does not exist in group `{group}`.')
+
+
+@pytest.fixture
+def entry_points(monkeypatch) -> EntryPointManager:
+    """Return an instance of the ``EntryPointManager`` which allows to temporarily add or remove entry points.
+
+    This fixture creates a deep copy of the entry point cache returned by the :func:`aiida.plugins.entry_point.eps`
+    method and then monkey patches that function to return the deepcopy. This ensures that the changes on the entry
+    point cache performed during the test through the manager are undone at the end of the function scope.
+    """
+    with warnings.catch_warnings():
+        warnings.filterwarnings('ignore', category=DeprecationWarning)
```

Comment:
Is this what `@suppress_deprecations` decorator do, why can't be used here?

Reply:
Because if I apply this decorator to the fixture itself, it doesn't just suppress the warnings when I call `eps_copy = copy.deepcopy(module_entry_point.eps())`, but also for the entire scope of the fixture. This includes the run time of the test that uses the fixture, and so it would essentially capture *all* warnings, even those emitted during the  test itself. But those we actually *do* want to see.

I can add a comment in the docstring why I cannot use the decorator for the fixture.

Reply:
I added the note to the docstring